tests/unit/dataactbroker/test_fileRoutes.py

```diff
@@ -60,3 +61,85 @@ def test_list_submissions(file_app, database, user_constants, job_constants):
+                       .filter_by(name='award_financial').one(), job_type=csv_validation, number_of_errors=0,
+                       file_size=123, job_status=finished_job)
+    job_d1 = JobFactory(submission_id=sub.submission_id, file_type=database.session.query(FileType)
+                        .filter_by(name='award_procurement').one(), job_type=csv_validation, number_of_errors=0,
+                        file_size=123, job_status=finished_job)
+    job_d2 = JobFactory(submission_id=sub.submission_id, file_type=database.session.query(FileType)
+                        .filter_by(name='award').one(), job_type=csv_validation, number_of_errors=0, file_size=123,
+                        job_status=finished_job)
+    job_e = JobFactory(submission_id=sub.submission_id, file_type=database.session.query(FileType)
+                       .filter_by(name='executive_compensation').one(), job_type=csv_validation, number_of_errors=0,
+                       file_size=123, job_status=finished_job)
+    job_f = JobFactory(submission_id=sub.submission_id, file_type=database.session.query(FileType)
+                       .filter_by(name='sub_award').one(), job_type=csv_validation, number_of_errors=0, file_size=123,
+                       job_status=finished_job)
+    job_cross_file = JobFactory(submission_id=sub.submission_id, file_type=None, job_type=validation,
+                                number_of_errors=0, file_size=123, job_status=finished_job)
+
+    database.session.add_all([job_a, job_b, job_c, job_d1, job_d2, job_e, job_f, job_cross_file])
+    database.session.commit()
+
+    # Everything ok
+    response = file_app.get("/v1/check_current_page/?submission_id=" + str(sub.submission_id))
+    response_json = json.loads(response.data.decode('UTF-8'))
+    assert response_json['step'] == '5'
+
+    job_e.job_status_id = 6
+    database.session.commit()
+    # E or F failed
+    response = file_app.get("/v1/check_current_page/?submission_id=" + str(sub.submission_id))
+    response_json = json.loads(response.data.decode('UTF-8'))
+    assert response_json['step'] == '4'
+
+    job_cross_file.job_status_id = 6
+    database.session.commit()
+    # Fail C file upload
+    response = file_app.get("/v1/check_current_page/?submission_id=" + str(sub.submission_id))
+    response_json = json.loads(response.data.decode('UTF-8'))
+    assert response_json['step'] == '3'
+
+    job_d1.number_of_errors = 6
+    database.session.commit()
+    # Fail C file upload
+    response = file_app.get("/v1/check_current_page/?submission_id=" + str(sub.submission_id))
+    response_json = json.loads(response.data.decode('UTF-8'))
+    assert response_json['step'] == '2'
+
+    job_c.number_of_errors = 6
+    database.session.commit()
+    # Fail C file upload
+    response = file_app.get("/v1/check_current_page/?submission_id=" + str(sub.submission_id))
+    response_json = json.loads(response.data.decode('UTF-8'))
+    assert response_json['step'] == '1'
```

Comment:
We need a test somewhere in here (perhaps at the end) where E/F have successfully generated but it still sends us back to an earlier point because something earlier failed (cross-file will not be status 4 if it or anything before it failed).

We also need a test where A/B/C were reuploaded and failed (cross-file will be 1 in this case) and D files were already generated. It needs to go back to the first step in this case